tests/network.md

```diff
@@ -403,6 +403,53 @@ If the fork itself fails, we still close the connection:
 Exception: Failure "Simulated error".
 ```
 
+## run_server
+
+```ocaml
+# let test_run_server env sw =
+  let connection_handler flow _addr =
+    traceln "Server accepted connection from client";
+    Fun.protect
+      (fun () ->
+         let msg = read_all flow in
+         traceln "Server received: %S" msg;
+      ) 
+      ~finally:(fun () -> Eio.Flow.copy_string "Bye" flow)
+  in
+  let server_sock = Eio.Net.listen ~reuse_addr:true ~backlog:128 ~sw env#net addr in
+  let server () = 
+    Eio.Net.run_server ~max_connections:10 ~additional_domains:(env#domain_mgr, 10) ~on_error:raise ~sw
+      server_sock connection_handler
+  in
+  server ();;
+val test_run_server :
+  < domain_mgr : #Eio__.Domain_manager.t; net : #Eio.Net.t; .. > ->
+  Switch.t -> unit = <fun>
+
+# let run_client env sw =
+  let flow = Eio.Net.connect ~sw env#net addr in
+  Eio.Flow.copy_string "Hello from client" flow;
+  Eio.Flow.shutdown flow `Send;
+  let msg = read_all flow in
+  traceln "client received: %S" msg;;
+val run_client : < net : #Eio.Net.t; .. > -> Switch.t -> unit = <fun>
+```
+
+```ocaml
+# Eio_main.run @@ fun env ->
+  try
+    Switch.run @@ fun sw ->
+    Fiber.all 
+      [(fun () -> test_run_server env sw)
+      ;(fun () -> run_client env sw; Switch.fail sw Graceful_shutdown)
```

Comment:
It seems doing below doesn't invoke the error handler as it does in `accept_fork`
```
  let on_error = function
    | Graceful_shutdown -> print_string "graceful"; ()
    | ex -> traceln "Error handling connection: %s" (Printexc.to_string ex)
  in
```

Reply:
I have changed to raising the exn rather than failing the switch along the lines of

```
let test_address addr ~net sw =
  let server = Eio.Net.listen net ~sw ~reuse_addr:true ~backlog:5 addr in
  Fiber.both
    (fun () -> run_server ~sw server)
    (fun () ->
      run_client ~sw ~net ~addr;
      traceln "Client finished - cancelling server";
      raise Graceful_shutdown
    )
```